libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/BannerInstructionEvent.kt

```diff
@@ -1,34 +1,58 @@
 package com.mapbox.navigation.core.trip.session
 
 import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.navigation.utils.internal.ifNonNull
 
-internal class BannerInstructionEvent {
+internal class BannerInstructionEvent private constructor() {
 
-    var latestInstructionIndex: Int? = null
+    var latestInstructionWrapper: LatestInstructionWrapper? = null
         private set
 
+    val latestInstructionIndex: Int?
```

Comment:
You use this getter only in one place in `MapboxTripSession` and it's used together with `latestBannerInstructions`. You should get the wrapper there in order to guarantee that the values are synced.

Reply:
I didn't want to refactor it a lot to avoid huge changes. The wrapper was added to support removing the latest instructions first

Reply:
> to avoid huge changes

Which huge changes? I mean:
```
Index: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt b/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt
--- a/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt	(revision 8766f0970a64ade7bcf80a3bfe0341ef9337529c)
+++ b/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt	(date 1670249275785)
@@ -368,12 +368,13 @@
                 )
             }
             val remainingWaypoints = tripStatus.calculateRemainingWaypoints()
+            val instructionsWrapper = bannerInstructionEvent.latestInstructionWrapper
             val routeProgress = getRouteProgressFrom(
                 tripStatus.route,
                 tripStatus.navigationStatus,
                 remainingWaypoints,
-                bannerInstructionEvent.latestBannerInstructions,
-                bannerInstructionEvent.latestInstructionIndex,
+                instructionsWrapper?.latestBannerInstructions,
+                instructionsWrapper?.latestInstructionIndex,
                 lastVoiceInstruction
             ).also {
                 if (it == null) {
```